File: General/OpenSave/vb/OpenSave/OpenSaveSettings.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

<Serializable> _
Public Class OpenSaveSettings
    Private _applicationSettings As New ApplicationSettings()
    Private _assemblySettings As New AssemblySettings()
    Private _draftSettings As New DraftSettings()
    Private _partSettings As New PartSettings()
    Private _sheetMetalSettings As New SheetMetalSettings()
    Private _weldmentSettings As New WeldmentSettings()

    Public Sub New()
        Application = New ApplicationSettings()
        Assembly = New AssemblySettings()
        Draft = New DraftSettings()
        Part = New PartSettings()
        SheetMetal = New SheetMetalSettings()
        Weldment = New WeldmentSettings()
    End Sub

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property Application() As ApplicationSettings

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property Assembly() As AssemblySettings

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property Draft() As DraftSettings

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property Part() As PartSettings

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property SheetMetal() As SheetMetalSettings

    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public Property Weldment() As WeldmentSettings
End Class

<Serializable> _
Public Class ApplicationSettings
    Public Sub New()
        Visible = True
    End Sub

    Public Property DisableAddins() As Boolean
    Public Property DisplayAlerts() As Boolean
    Public Property Visible() As Boolean

    Public Overrides Function ToString() As String
        Return "Application Settings"
    End Function
End Class

<Serializable> _
Public Class AssemblySettings
    Public Overrides Function ToString() As String
        Return "Assembly Settings"
    End Function
End Class

<Serializable> _
Public Class DraftSettings
    Public Property UpdateDrawingViews() As Boolean

    Public Overrides Function ToString() As String
        Return "Draft Settings"
    End Function
End Class

<Serializable> _
Public Class PartSettings
    Public Overrides Function ToString() As String
        Return "Part Settings"
    End Function
End Class

<Serializable> _
Public Class SheetMetalSettings
    Public Overrides Function ToString() As String
        Return "SheetMetal Settings"
    End Function
End Class

<Serializable> _
Public Class WeldmentSettings
    Public Overrides Function ToString() As String
        Return "Weldment Settings"
    End Function
End Class

File: General/OpenSave/cs/OpenSave/OpenSaveSettings.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace SolidEdge.OpenSave
{
    [Serializable]
    public class OpenSaveSettings
    {
        private ApplicationSettings _applicationSettings = new ApplicationSettings();
        private AssemblySettings _assemblySettings = new AssemblySettings();
        private DraftSettings _draftSettings = new DraftSettings();
        private PartSettings _partSettings = new PartSettings();
        private SheetMetalSettings _sheetMetalSettings = new SheetMetalSettings();
        private WeldmentSettings _weldmentSettings = new WeldmentSettings();

        public OpenSaveSettings()
        {
            Application = new ApplicationSettings();
            Assembly = new AssemblySettings();
            Draft = new DraftSettings();
            Part = new PartSettings();
            SheetMetal = new SheetMetalSettings();
            Weldment = new WeldmentSettings();
        }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public ApplicationSettings Application { get; set; }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public AssemblySettings Assembly { get; set; }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public DraftSettings Draft { get; set; }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public PartSettings Part { get; set; }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public SheetMetalSettings SheetMetal { get; set; }

        [TypeConverter(typeof(ExpandableObjectConverter))]
        public WeldmentSettings Weldment { get; set; }
    }

    [Serializable]
    public class ApplicationSettings
    {
        public ApplicationSettings()
        {
            Visible = true;
        }

        public bool DisableAddins { get; set; }
        public bool DisplayAlerts { get; set; }
        public bool Visible { get; set; }

        public override string ToString()
        {
            return "Application Settings";
        }
    }

    [Serializable]
    public class AssemblySettings
    {
        public override string ToString()
        {
            return "Assembly Settings";
        }
    }

    [Serializable]
    public class DraftSettings
    {
        public bool UpdateDrawingViews { get; set; }

        public override string ToString()
        {
            return "Draft Settings";
        }
    }

    [Serializable]
    public class PartSettings
    {
        public override string ToString()
        {
            return "Part Settings";
        }
    }

    [Serializable]
    public class SheetMetalSettings
    {
        public override string ToString()
        {
            return "SheetMetal Settings";
        }
    }

    [Serializable]
    public class WeldmentSettings
    {
        public override string ToString()
        {
            return "Weldment Settings";
        }
    }
}
